### DISCUSSION TITLE
How to update the translation into the project by clicking Publish?

### DISCUSSION THREAD

**Discussion by ChengQuanYe** [crowdin/mobile-sdk-ios#244] [Q&A] [answered]
Why is it useless if the project integrates the SDK? How to update the translation into the project by clicking Publish?

**Comment by andrii-bodnar** [comment]
@ChengQuanYe could you please clarify your question? Thank you.

**Comment by ChengQuanYe** [comment]
> @ChengQuanYe could you please clarify your question? Thank you.

<img width="691" alt="WechatIMG172" src="https://github.com/crowdin/mobile-sdk-ios/assets/17982983/632e8372-53a6-4c75-9336-8d6cea199db5">
<img width="1061" alt="WechatIMG173" src="https://github.com/crowdin/mobile-sdk-ios/assets/17982983/e8048088-a2b1-490e-8204-ed30f48b4871">
<img width="688" alt="WechatIMG174" src="https://github.com/crowdin/mobile-sdk-ios/assets/17982983/d65a1b22-84b4-44da-9106-0b4e3f2fb456">

I followed the steps to integrate the SDK, but the translated strings file was not generated. Which step did I take wrong?

**Comment by andrii-bodnar** [comment]
@ChengQuanYe the integration doesn't generate the localization file. Please refer to the official documentation to get more details on the matter - https://crowdin.github.io/mobile-sdk-ios/setup

Feel free to check out the [example project](https://crowdin.github.io/mobile-sdk-ios/example). This project illustrates how you can use the Crowdin SDK features with a real iOS app.

Also, make sure you've translated your strings in Crowdin to the target language in which you're testing the translation delivery in the app.

**Comment by ChengQuanYe** [comment]
> @ChengQuanYe the integration doesn't generate the localization file. Please refer to the official documentation to get more details on the matter - https://crowdin.github.io/mobile-sdk-ios/setup
> 
> Feel free to check out the [example project](https://crowdin.github.io/mobile-sdk-ios/example). This project illustrates how you can use the Crowdin SDK features with a real iOS app.
> 
> Also, make sure you've translated your strings in Crowdin to the target language in which you're testing the translation delivery in the app.

Thank you. Can this SDK automatically update the .strings file copy in the project? Or do you need to copy and paste the translated .strings file into the project after crowdin translation?

**Comment by andrii-bodnar** [comment, marked as answer]
@ChengQuanYe this integration doesn't do that, it only delivers the translations to the application over-the-air.

There are a lot of ways to manage source and translated files in your project. for example:

- [Crowdin GitHub integration](https://support.crowdin.com/github-integration/)
- [Crowdin GitHub Action](https://github.com/crowdin/github-action)
- [Crowdin CLI](https://crowdin.github.io/crowdin-cli/)

Visit the [Crowdin Store](https://store.crowdin.com/) for more integrations.
